cash-register/cash-register.js

```diff
@@ -0,0 +1,129 @@
+const cashStatus = {
+  insufficientFunds: "INSUFFICIENT_FUNDS",
+  closed: "CLOSED",
+  open: "OPEN"
+}
+
+function checkCashRegister(price, cash, cid) {
+  var change = {
+    status: "",
+    change: []
+  };
+
+  const cashInDraw = getTotalCashInDraw(cid);
+  const changeToReturn = cash - price;
+  change.status = getStatus(changeToReturn, cashInDraw);
+  if (change.status === cashStatus.insufficientFunds) {
+    return change;
+  }
+  if (change.status === cashStatus.closed) {
+    change.change = cid
+  } else {
+    let changeArrayForOpenStatus = getChangeForOpenStatus(cid, changeToReturn);
+    if (changeArrayForOpenStatus.length > 0) {
+      change.change = changeArrayForOpenStatus;
+    } else {
+      change.status = cashStatus.insufficientFunds;
+    }
+  }
+  return change;
+}
+
+function getChangeForOpenStatus(cid, changeToReturn) {
+  let currencyArray = [];
+  const currencyRate = [{
+      name: 'ONE HUNDRED',
+      val: 100.00
+    },
+    {
+      name: 'TWENTY',
+      val: 20.00
+    },
+    {
+      name: 'TEN',
+      val: 10.00
+    },
+    {
+      name: 'FIVE',
+      val: 5.00
+    },
+    {
+      name: 'ONE',
+      val: 1.00
+    },
+    {
+      name: 'QUARTER',
+      val: 0.25
+    },
+    {
+      name: 'DIME',
+      val: 0.10
+    },
+    {
+      name: 'NICKEL',
+      val: 0.05
+    },
+    {
+      name: 'PENNY',
+      val: 0.01
+    }
+  ];
+  cid = sortCashInDrawer(cid, currencyRate);
+  console.log(cid);
+  let change = [];
+  let value = 0;
+  let count = 0;
+  while (count < cid.length) {
```

Comment:
create variables for currencyValue = cid[count][1] and currencyName = cid[count][0]